src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -379,6 +380,16 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     {
                         _watchedSettings = watchedSettings;
 
+                        Dictionary<string, ConfigurationSetting> precedenceData = OrderDataByPrecedence(out Dictionary<string, string> logIdentifiers);
```

Comment:
Why dont we directly pass `logInfoBuilder` here? Is it because we want to rearrange the logging order depending on the precedence order?

Reply:
Its generally not a good practice to have a method that returns a value and also has out parameters.

Reply:
This is only reordering logs for feature flags - if we are reordering, I think we should do it for key-values too.

Alternatively, instead of rearranging logs, we log everything as it happens but while preparing data, we add an additional log statement for all overwrites of changed settings. I'm thinking of something like this:

During RefreshAsync:
`Key-value read from App Configuration. Change:'Modified' Key:'FontColor' Label:'Dev' Endpoint:'contoso.azconfig.io'`
`Setting updated. Key: FontColor`

During PrepareData:
`Value of Key:'FontColor' Label:'Prod' will override previous values.`

I think we dont need the new helper method for ordering precedence data, it can be part of `PrepareData`, and `PrepareData` can be updated to accept a list of key value changes:

```
private async Task<Dictionary<string, string>> PrepareData(IEnumerable<KeyValueChange> changes, CancellationToken cancellationToken = default)
```

Notice that we dont really need to pass a dictionary of data to PrepareData because its always operating on `_mappedData`.

Reply:
Yea, I see now that I missed reordering the info level logs for key-values.

I mostly understand and I'm willing to change it, but I think this log feels like it could be confusing because of the order. Did you suggest it because the log rearranging adds some iteration that we don't necessarily need?

Reply:
I'm trying to think of a way to do it without having to pass around a new placeholder data structure - logInfoBuilder is itself a placeholder, now we're passing around a placeholder for a placeholder :)
It also feels odd to reorder info logs, but not debug logs. So either we do both or neither.
If my previous suggestion feels confusing, how about getting rid of the logInfoBuilder and logDebugBuilder string builders and just pass around `cachedInfoLogs` and `cachedDebugLogs`?

Reply:
Makes sense, my original thought was that the debug messages didn't really need to be reordered since they just mention reading key-values so I didn't want to add `cachedDebugLogs`. I agree that it looks odd to use a different variable for each log level, but I also think we should only add `cachedDebugLogs` if we want to reorder the debug logs, which I don't see a reason for right now. I know that didn't help much, but that's just how I see it right now.

Reply:
Do you mind giving a sample of what the debug and info logs will look like with the changes you currently have?

Reply:
Code to select feature flags: 
```
 options.UseFeatureFlags(ffoptions =>
 {
     ffoptions.Select(KeyFilter.Any, null);
     ffoptions.Select(KeyFilter.Any, "prod");
 });
```

Changing flag `alpha` with label `null`:
```
dbug: Microsoft.Extensions.Configuration.AzureAppConfiguration.Refresh[0]
      Feature flag read from App Configuration. Key:'alpha' Label:'' Endpoint:'https://ajusupovic-ac2.azconfig.io'
```

Changing flag `alpha` with label `prod`:
```
dbug: Microsoft.Extensions.Configuration.AzureAppConfiguration.Refresh[0]
      Feature flag read from App Configuration. Key:'alpha' Label:'prod' Endpoint:'https://ajusupovic-ac2.azconfig.io'
info: Microsoft.Extensions.Configuration.AzureAppConfiguration.Refresh[0]
      Feature flag updated. Key:'alpha'
```

Reply:
Thanks for the example. The current code behaves differently though. Your first scenario will produce logs like the second scenario. Changing flag `alpha` with label `null` will still have an info log saying `Feature flag updated. Key:'alpha'`.